internal/policy/cache/cache.go

```diff
@@ -72,25 +70,19 @@ func (c *PolicyCache) Set(key string, value string, err error) {
 }
 
 // NewPolicyCache creates and returns a new PolicyCache instance.
-func NewPolicyCache(ctx context.Context) (*PolicyCache, error) {
+func NewPolicyCache(ctx context.Context) *PolicyCache {
 	cache, ok := ctx.Value(policyCacheKey).(*PolicyCache)
 	if ok && cache != nil {
-		return cache, nil
-	}
-
-	c, err := CreatePolicyCache()
-	if err != nil {
-		log.Debug("Failed to create PolicyCache")
-		return nil, err
+		return cache
 	}
 
-	return c, nil
+	return CreatePolicyCache()
```

Comment:
```suggestion
	return &PolicyCache{
		Data: sync.Map{},
	}
```
With this change you could get rid of the CreatePolicyCache entirely, i think it is used only here